numpy/_core/src/multiarray/unique.cpp

```diff
@@ -76,6 +86,11 @@ size_t hash_complex_clongdouble(const npy_clongdouble *value, npy_bool equal_nan
     #if defined(HAVE_LDOUBLE_INTEL_EXTENDED_12_BYTES_LE) || \
         defined(HAVE_LDOUBLE_INTEL_EXTENDED_16_BYTES_LE) || \
         defined(HAVE_LDOUBLE_MOTOROLA_EXTENDED_12_BYTES_BE)
+
+    // Get rid of -0.0.
+    value_real = (value_real == 0.0) ? 0.0 : value_real;
+    value_imag = (value_imag == 0.0) ? 0.0 : value_imag;
```

Comment:
```suggestion
    value_real = (value_real == 0.0) ? NPY_PZEROL : value_real;
    value_imag = (value_imag == 0.0) ? NPY_PZEROL : value_imag;
```
Using NPY_PZEROL macro may be better.